src/Mvc/Mvc.NewtonsoftJson/src/DependencyInjection/NewtonsoftJsonMvcOptionsSetup.cs

```diff
@@ -59,33 +59,58 @@ internal class NewtonsoftJsonMvcOptionsSetup : IConfigureOptions<MvcOptions>
 
         public void Configure(MvcOptions options)
         {
-            options.OutputFormatters.Add(new NewtonsoftJsonOutputFormatter(_jsonOptions.SerializerSettings, _charPool));
+            InsertBefore(
+                options.OutputFormatters,
+                "Microsoft.AspNetCore.Mvc.Formatters.JsonOutputFormatter",
+                new NewtonsoftJsonOutputFormatter(_jsonOptions.SerializerSettings, _charPool));
 
             // Register JsonPatchInputFormatter before JsonInputFormatter, otherwise
             // JsonInputFormatter would consume "application/json-patch+json" requests
             // before JsonPatchInputFormatter gets to see them.
             var jsonInputPatchLogger = _loggerFactory.CreateLogger<NewtonsoftJsonPatchInputFormatter>();
-            options.InputFormatters.Add(new NewtonsoftJsonPatchInputFormatter(
+            var jsonPatchInputFormatter = new NewtonsoftJsonPatchInputFormatter(
                 jsonInputPatchLogger,
                 _jsonOptions.SerializerSettings,
                 _charPool,
                 _objectPoolProvider,
                 options,
-                _jsonOptions));
+                _jsonOptions);
+
+            const string JsonInputFormatter = "Microsoft.AspNetCore.Mvc.Formatters.JsonInputFormatter";
+
+            InsertBefore(options.InputFormatters, JsonInputFormatter, jsonPatchInputFormatter);
 
             var jsonInputLogger = _loggerFactory.CreateLogger<NewtonsoftJsonInputFormatter>();
-            options.InputFormatters.Add(new NewtonsoftJsonInputFormatter(
+            var jsonInputFormatter = new NewtonsoftJsonInputFormatter(
                 jsonInputLogger,
                 _jsonOptions.SerializerSettings,
                 _charPool,
                 _objectPoolProvider,
                 options,
-                _jsonOptions));
+                _jsonOptions);
 
+            InsertBefore(options.InputFormatters, JsonInputFormatter, jsonInputFormatter);
```

Comment:
We should remove the other ones too IMO. It's OK if we have to change what the method is called to make that clear. I think users want one thing or the other.